### ISSUE TITLE
W1b: `bbox_generator3d` builds boxes one wider than requested per axis, unlike `bbox_generator`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4018]
`bbox_generator` places the far corner at `start + size - 1` (inclusive extent = `size`, consistent with `infer_bbox_shape`'s `+1`), while `bbox_generator3d` places it at `start + size` (`bbox.py:405-411`: `bbox[:, 1, 0] += width` etc.), so the 3D box measured by the module's own `infer_bbox_shape3d` is `size + 1` on every axis. The function's doctest already shows it: depths 30 and 60 in, `[31, 61]` out.

## Repro

```python
import torch
from kornia.geometry.bbox import bbox_generator, bbox_generator3d, infer_bbox_shape, infer_bbox_shape3d, bbox_to_mask, bbox_to_mask3d

one = torch.tensor([1])
g2 = bbox_generator(one, one, torch.tensor([3]), torch.tensor([2]))
g3 = bbox_generator3d(one, one, one, torch.tensor([3]), torch.tensor([2]), torch.tensor([2]))
print("2D width=3: x corners", sorted(set(g2[0, :, 0].tolist())), " measured w", infer_bbox_shape(g2.float())[1].tolist(), " mask px", bbox_to_mask(g2.float(), 8, 8).sum().item())
print("3D width=3: x corners", sorted(set(g3[0, :, 0].tolist())), " measured w", infer_bbox_shape3d(g3)[2].tolist(), " mask px", bbox_to_mask3d(g3.float(), (8, 8, 8)).sum().item())
```

## Observed

```
2D width=3: x corners [1, 3]  measured w [3.0]  mask px 6.0      <- 3 x 2
3D width=3: x corners [1, 4]  measured w [4]    mask px 36.0     <- 4 x 3 x 3, asked for 3 x 2 x 2
```

## Expected

The 3D generator uses `start + size - 1` like the 2D one, so `bbox_generator3d(..., width=w)` has measured width `w`. Every current 3D caller (`kornia/augmentation/random_generator/_3d/crop.py`, `geometry/transform/crop3d.py`) gets boxes one voxel smaller per axis after the fix, and the doctest literal changes — a semantic change → window, pinned as a wart until then.

## Classification

**Semantic contract / 2D-3D split.** Idiom 3 (the computed extent does not match the documented `width`/`height`/`depth` contract).

## Disposition

**MERGE IN WINDOW — behavior change.** Canonical: `bbox_generator`'s `start + size - 1` (the form that agrees with `infer_bbox_shape*`). Shape: align contract, both names kept. Divergence today (executed): width 3 → measured 3 in 2D, 4 in 3D. This is the behavior-change discussion CONTRIBUTING asks for, and the decision is: **in the coordinated Repair & Deprecation window, not before. A PR that aliases or consolidates these ahead of it will be closed and pointed here.**

## Environment

kornia 0.9.0rc1 @ `0685b1f5` (main), torch 2.9.1, python 3.11, cpu. Found during the bbox/boxes conventions audit (batch 4, tracking issue #4020); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
